rule/rule.go

```diff
@@ -238,6 +238,25 @@ func MatchBuildFileName(dir string, names []string, files []os.FileInfo) string
 	return ""
 }
 
+// SyncMacroFile syncs the file's syntax tree with another file's. This is
+// useful for keeping multiple macro definitions from the same .bzl file in sync.
+func (f *File) SyncMacroFile(from *File) {
```

Comment:
Let's be very careful here. Having multiple files share an AST means there are multiple mutable references to the same data structure. That could result in some subtle bugs if other edits are made and `Sync` is called later.

Ideally, I think this method would copy the macro from one `*File` to another, replacing the function of the same name that was already there. I'm not sure if there's a way to copy ASTs, so adding or replacing the macro function in `f.File.Stmt` is probably good enough.